File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>
#include <vector>
#include <cstring>
#define FASTIO  cin.tie(0); cout.tie(0); ios_base::sync_with_stdio(false);
#define INTMAX 1000000001
using namespace std;

int main() {
    FASTIO;
    queue<int> q;
    
    int tc;
    cin >> tc;
    
    for(int i = 0; i < tc; i++){
        string op;
        int num;
        
        cin >> op;
        
        if(op == "push"){
            cin >> num;
            q.push(num);
        }
        
        if(op == "pop"){
            if(q.size() == 0) {cout << -1 << '\n';}
            else {
                cout << q.front() << '\n';
                q.pop();
            }
        }
        
        if(op == "size"){
            cout << q.size() << '\n';
        }
        
        if(op == "empty"){
            cout << q.empty()<< '\n';
        }
        
        if(op == "front"){
            if(q.size() == 0) {cout << -1 << '\n';}
            else {
                cout << q.front()<< '\n';
            }
        }
        
        if(op == "back"){
            if(q.size() == 0) {cout << -1 << '\n';}
            else {
                cout << q.back()<< '\n';
            }
        }
        
    }
   
    return 0;
}

File: 백준/Silver/18258. 큐 2/큐 2.js
const fs = require("fs");
const input = fs
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .split("\n");
const tc = Number(input[0]);

class Queue {
  constructor() {
    this.data = [];
    this.head = 0;
    this.tail = 0;
  }

  push(value) {
    this.data[this.tail++] = value;
  }

  size() {
    return this.tail - this.head;
  }

  pop() {
    if (this.size() === 0) return -1;
    const value = this.data[this.head];
    this.head++;
    return value;
  }

  front() {
    return this.size() === 0 ? -1 : this.data[this.head];
  }

  back() {
    return this.size() === 0 ? -1 : this.data[this.tail - 1]; // 수정
  }

  empty() {
    return this.size() === 0 ? 1 : 0;
  }
}

let q = new Queue();
let answer = [];

for (let i = 1; i <= tc; i++) { // 반복문 범위 수정
  const [op, num] = input[i].split(" ");

  switch (op) {
    case "push":
      q.push(Number(num));
      break;

    case "pop":
      answer.push(q.pop());
      break;

    case "size":
      answer.push(q.size());
      break;

    case "empty":
      answer.push(q.empty());
      break;

    case "front":
      answer.push(q.front());
      break;

    case "back":
      answer.push(q.back());
      break;
  }
}

// 모든 결과를 한 번에 출력
console.log(answer.join("\n"));